elobot.py

```diff
@@ -54,245 +30,233 @@ def get_channel_id(self, channel_name):
 
-        winner_elo_delta, loser_elo_delta = self.apply_match(match)
-        self.talk_to(match.winner_handle, f'Your new ELO is {self.players[match.winner_handle].rating} (+{winner_elo_delta}).')
-        self.talk_to(match.loser_handle , f'Your new ELO is {self.players[match.loser_handle ].rating} ({loser_elo_delta }).')
+        if not players:
+            if verbose:
+                self.talk_to(user_handle, f'Cannot confirm match #{match_id}! You\'re not in it!')
+            return False
 
-    def delete(self, user_handle, match_id):
-        match = self.get_pending(match_id)
-        if not match: return
+        if not any(map(lambda p: p.pending, players)):
+            if verbose:
+                self.talk_to(user_handle, f'You have already confirmed match #{match_id}!')
+            return False
 
-        if match.winner_handle != user_handle:
-            self.talk_to(user_handle, f'You are not allowed to delete match #{match_id}!')
+        (Player.update(pending = False)
+            .where(Player.handle == user_handle, Player.match == match_id)
+            .execute())
 
-        match.delete_instance()
-        self.talk(f'Deleted match #{match_id}.')
+        if verbose:
+            self.talk_to(user_handle, f'Confirmed match #{match_id}!')
+
+        if not match.pending:
+            observe_match(match)
+            if verbose:
+                self.flush_elo_cache()
+        return True
 
     def print_leaderboard(self):
         table = []
 
-        for slack_handle, player in sorted(self.players.items(), key=lambda p: p[1].rating, reverse=True):
-            win_streak = self.get_win_streak(slack_handle)
-            streak_text = '(won {} in a row)'.format(win_streak) if win_streak >= self.min_streak_len else ''
-            table.append([self.slack_client.get_name(slack_handle), player.rating, player.wins, player.losses, streak_text])
+        # TODO: Inefficient
+        for user_handle in set(map(lambda p: p.handle, Player.select())):
+            win_streak = get_streak(user_handle)
+            streak_text = 'Won {} in a row'.format(win_streak) if win_streak >= self.min_streak_len else '-'
+            name = self.slack_client.get_name(user_handle)
+            table.append([name, get_elo(user_handle), get_wins(user_handle), get_losses(user_handle), streak_text])
 
         self.talk('```' + tabulate(table, headers=['Name', 'ELO', 'Wins', 'Losses', 'Streak']) + '```')
 
     def print_unconfirmed(self):
         table = []
 
-        for match in Match.select().where(Match.pending == True).order_by(Match.played.desc()).limit(25):
-            match_played_utc = match.played.replace(tzinfo=from_zone)
-            match_played_pst = match_played_utc.astimezone(to_zone)
+        # TODO: There must be a cleaner way to do this line
+        for match in it.islice(filter(lambda m: m.pending, Match.select().order_by(Match.datetime.desc())), 0, 25):
```

Comment:
yeah pretty sure you can `select('where pending == true)` or something

Reply:
Does that work, since `pending` is a method rather than a field?

Reply:
oh right, you swapped it